core/src/main/java/org/pentaho/di/connections/ConnectionProvider.java

```diff
@@ -44,4 +45,6 @@ public interface ConnectionProvider<T extends ConnectionDetails> {
   boolean test( T connectionDetails ) throws KettleException;
 
   T prepare( T connectionDetails ) throws KettleException;
+
+  void hydrate( T connectionDetails );
```

Comment:
Please, add a doclet with something like `Initializes a new connection instance, or one deserialized from the meta-store.`